features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/route/engine/condition/engine/ShardingConditionEngine.java

```diff
@@ -21,11 +21,22 @@
 import org.apache.shardingsphere.infra.binder.statement.SQLStatementContext;
 
 import java.util.List;
+import org.apache.shardingsphere.infra.metadata.database.ShardingSphereDatabase;
+import org.apache.shardingsphere.infra.util.spi.type.required.RequiredSPI;
+import org.apache.shardingsphere.sharding.rule.ShardingRule;
 
 /**
  * Sharding condition engine.
  */
-public interface ShardingConditionEngine<T extends SQLStatementContext<?>> {
+public interface ShardingConditionEngine<T extends SQLStatementContext<?>> extends RequiredSPI {
+    
+    /**
+     * Initialize the sharding condition engine.
+     *
+     * @param rule sharding rule
+     * @param database sharding database
+     */
+    void init(ShardingRule rule, ShardingSphereDatabase database);
```

Comment:
Why add init method?

Reply:
@strongduanmu The Java `ServiceLoader` implementation requires a class with a default constructor, so we cannot have a parameterized constructor `public DefaultShardingConditionEngine(final ShardingRule rule, final ShardingSphereDatabase database)`.  The method `void init(ShardingRule rule, ShardingSphereDatabase database)` is added to allow the implementation class to prepare its internal structures.

Reply:
Thank you for your explaination.